llvm/lib/Target/AMDGPU/AMDGPULegalizerInfo.cpp

```diff
@@ -5805,6 +5806,10 @@ Register AMDGPULegalizerInfo::fixStoreSourceType(
   if (hasBufferRsrcWorkaround(Ty))
     return castBufferRsrcToV4I32(VData, B);
 
+  if (shouldBitcastLoadStoreType(ST, Ty, MemTy) || Ty.isPointerVector()) {
```

Comment:
I think the Ty.isPointerVector() is redundant, these should be handled under the loadStoreBitcastWorkaround case

Reply:
It isn't. All the pointers vectors fall to the `Size <= 64` case and so don't get bitcast

Reply:
This sounds like just a bug, and this should be folded into shouldBitcastLoadStoreType. It already tries to handle this case, just too late

Reply:
Except that a regular G_LOAD or G_STORE can handle vectors of pointers just fine, and moving the condition changes a bunch of tests

Reply:
I don't think they do work fine. The buffer cases should behave identically to G_LOAD and G_STORE, and both currently rely on the bitcast legalization.

Based on the 64-bit check, the 128-bit <2 x ptr> cases work. The 64-bit cases, e.g. <2 x ptr addrspace(3)>  fail:
https://godbolt.org/z/4vWacff1d

So yes, tests should change and start working